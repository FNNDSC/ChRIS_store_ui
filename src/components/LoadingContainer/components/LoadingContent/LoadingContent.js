import React from 'react';
import PropTypes from 'prop-types';

const LoadingContent = (props) => {
  const computedStyle = {
    width: props.width,
    height: props.height,
    marginTop: props.top,
    marginLeft: props.left,
    marginBottom: props.bottom,
    marginRight: props.right,
  };

  let addedClasses = props.className;
  switch (props.type) {
    case 'white':
      addedClasses += ' white';
      break;
    default:
  }

  return (
    <div
      className={`loading ${addedClasses}`}
      style={computedStyle}
    />
  );
};

LoadingContent.propTypes = {
  className: PropTypes.string,
  type: PropTypes.string,
  width: PropTypes.string.isRequired,
  height: PropTypes.string.isRequired,
  top: PropTypes.string,
  left: PropTypes.string,
  bottom: PropTypes.string,
  right: PropTypes.string,
};

LoadingContent.defaultProps = {
  className: '',
  type: '',
  top: '0',
  left: '0',
  bottom: '0',
  right: '0',
};

export default LoadingContent;
