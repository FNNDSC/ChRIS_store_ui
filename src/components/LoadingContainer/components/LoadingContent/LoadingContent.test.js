import React from 'react';
import { shallow } from 'enzyme';
import LoadingContent from './LoadingContent';

describe('LoadingContent', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<LoadingContent width="" height="" />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render loading div', () => {
    expect(wrapper.find('div.loading')).toHaveLength(1);
  });

  it('should render width and height props in CSS', () => {
    wrapper.setProps({
      width: '314px',
      height: '159px',
    });

    const styleProp = wrapper.find('div.loading').prop('style');
    expect(styleProp.width).toEqual('314px');
    expect(styleProp.height).toEqual('159px');
  });

  it('should render className prop', () => {
    wrapper.setProps({
      className: 'testClass',
    });

    expect(wrapper.find('div.loading.testClass')).toHaveLength(1);
  });

  it('should render type prop', () => {
    wrapper.setProps({
      className: 'shouldHaveThisAlso',
      type: 'white',
    });

    expect(wrapper
      .find('div.loading.white.shouldHaveThisAlso'))
      .toHaveLength(1);
  });

  it('should render margin props in CSS', () => {
    wrapper.setProps({
      top: '265px',
      left: '358px',
      bottom: '979px',
      right: '323px',
    });

    const styleProp = wrapper.find('div.loading').prop('style');
    expect(styleProp.marginTop).toEqual('265px');
    expect(styleProp.marginLeft).toEqual('358px');
    expect(styleProp.marginBottom).toEqual('979px');
    expect(styleProp.marginRight).toEqual('323px');
  });
});
