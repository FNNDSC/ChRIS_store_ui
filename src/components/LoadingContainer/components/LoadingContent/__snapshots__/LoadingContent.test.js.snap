// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingContent should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingContent
    bottom="0"
    className=""
    height=""
    left="0"
    right="0"
    top="0"
    type=""
    width=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "loading ",
      "style": Object {
        "height": "",
        "marginBottom": "0",
        "marginLeft": "0",
        "marginRight": "0",
        "marginTop": "0",
        "width": "",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "loading ",
        "style": Object {
          "height": "",
          "marginBottom": "0",
          "marginLeft": "0",
          "marginRight": "0",
          "marginTop": "0",
          "width": "",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
